packages-tests/Caching/Detector/ChangedFilesDetectorTest.php

```diff
@@ -17,7 +21,26 @@ protected function setUp(): void
     {
         parent::setUp();
 
-        $this->changedFilesDetector = $this->getService(ChangedFilesDetector::class);
+        $smartFileSystem = $this->getService(SmartFileSystem::class);
+
+        $cacheDir = sys_get_temp_dir() . '/rector_changed_files_detector_test';
+        $fileCacheStorage = new FileCacheStorage(
+            $cacheDir,
+            $smartFileSystem
+        );
+        $cache = new Cache($fileCacheStorage);
+
+        // use a local object, instead of this registered service,
+        // so this unit test does not clear global ChangedFilesDetector caches
+        $this->changedFilesDetector = new ChangedFilesDetector(
+            new FileHashComputer(),
+            $cache
+        );
+    }
```

Comment:
This would require manual changes on dependency change

It might better to create standalone container and use it here to fetch the built service instead.

Reply:
> It might better to create standalone container and use it here to fetch the built service instead.

how todo that? is this done somewhere else already?

Reply:
Yes, the parent class is creating a container.